Doc/Manual/R.html

```diff
@@ -189,9 +189,46 @@ <H2><a name="R_nn7">33.7 Enumerations</a></H2>
 
 
 <p>
-enumerations are characters which are then converted back and forth to
-ints before calling the C routines. All of the enumeration code is
-done in R.
+R doesn't have a native enumeration type. Enumerations are represented
+as character strings in R, with calls to R functions that convert back
+and forth between integers.
+
+The details of enumeration names and contents are stored in hidden R
+environments, which are named according the the enumeration name - for
+example, an enumeration colour:
+<div class="code"><pre>
+enum colour { red=-1, blue, green = 10 };
+</pre></div>
+
+will be initialized by the following call in R:
+<div class="code"><pre>
+defineEnumeration("_colour",
+ .values=c("red" = .Call('R_swig_colour_red_get',FALSE, PACKAGE='enum_thorough'),
+"blue" = .Call('R_swig_colour_blue_get',FALSE, PACKAGE='enum_thorough'),
+"green" = .Call('R_swig_colour_green_get',FALSE, PACKAGE='enum_thorough')))
+
+</pre></div>
+
+which will create an environment named <tt>.__E___colour</tt>. The enumeration
+values are initialised via calls to C/C++ code, allowing complex
+values for enumerations to be used. Calls to the C/C++ code require
+the compiled library to be loaded, so a  <tt>delayedAssign</tt> is employed
+within <tt>defineEnumeration</tt> in order to allow the code to be easily used in R
+packages.
+
+The user typically does not need to access the enumeration lookup
+functions or know the name of the enumeration type used by
+R. Attributes containing the type information are attached by swig to
+functions requiring enumeration arguments or returning enumeration
+values, and those attributes are used to identify and access the
+appropriate environments and thus translate between characters
+and integers.
+
+The relevant functions, for debugging purposes, are <tt>enumToInteger</tt> and
+</tt>enumFromInteger</tt>.
+
+Anonymous enumerations are not supported.
```

Comment:
They generate code that compiles. In what way are they not supported? Are the symbols introduced by anonymous enums not available at all? Other languages wrap them as simple numeric values.

Reply:
As far as I know, the R module has always ignored them, but I will check further. There's certainly no direct way of implementing them via the R environment approach used by the rest of the enum framework.

Reply:
@wsfulton - regarding the cpp11 problems - I think the issue is that I need to do
```cpp
r_ans = Rf_ScalarInteger(static_cast<int>(*result)); 
```

However I presume I should only turn that on for cpp11 - what is the appropriate set of guards to use in `r.swg`?

Reply:
I have just confirmed that the old/original R enumeration framework ignored the anonymous enumerations and doesn't provide any way of accessing them. This PR, which is mostly about correctly initialising values for enumerations, doesn't change that. I will update the doc to be clearer.

Not sure at the moment how tricky it is to do something more useful - might be able to get a basic variable wrapper approach to do something.

Reply:
Okay, so anonymous enums ought to be wrapped by the equivalent to an R read-only integer variable / constant.

Reply:
The `const enum SWIGTYPE&` needs modifying with the cast. The cast can always be used and I've made a commit to fix this in 39e3c3b9d774a483fbec8f0fcf6c136833a79bf8. I'm merging this in shortly, so thanks for sorting this all out.